File: src/main/java/com/server/ordering/domain/dto/request/CouponDto.java
package com.server.ordering.domain.dto.request;

import com.server.ordering.domain.Coupon;
import com.server.ordering.domain.MyCoupon;
import lombok.*;

import static lombok.AccessLevel.PROTECTED;

@Getter @Setter
@RequiredArgsConstructor
@NoArgsConstructor(access = PROTECTED)
public class CouponDto {

    private Long couponId;
    @NonNull
    private String serialNumber;
    @NonNull
    private int value;

    public CouponDto(MyCoupon myCoupon) {
        this.couponId = myCoupon.getId();
        this.serialNumber = myCoupon.getSerialNumber();
        this.value = myCoupon.getValue();
    }
}

File: src/main/kotlin/com/byeoru/ordering_server/domain/dto/request/CouponDto.kt
package com.byeoru.ordering_server.domain.dto.request

import com.byeoru.ordering_server.domain.MyCoupon

class CouponDto {

    var couponId: Long? = null
    val serialNumber: String
    val value: Int

    constructor(myCoupon: MyCoupon) {
        couponId = myCoupon.id
        serialNumber = myCoupon.serialNumber
        value = myCoupon.value
    }

    constructor(serialNumber: String, value: Int) {
        this.serialNumber = serialNumber
        this.value = value
    }
}